### PR TITLE
5 hour session window usage info

### PR DESCRIPTION
**Pull request by EphemeralSapient**
fixes: #20 
fixes: #23 
Used claude code to implement that 5 hour session window thing, let me know if its okay

## Usage Examples

  ```bash
  # Show session usage and limits
  ccusage sessions

  # Filter by date range
  ccusage sessions --since 20250101 --until 20250131

  # Export as JSON for programmatic access
  ccusage sessions --json

  # Get session count programmatically
  ccusage sessions --json | jq '.sessions[0].totalSessions'
```

## Summary by CodeRabbit

- **New Features**
  - Introduced a "Session Windows Report" for the Claude Max plan, allowing users to view and track usage in 5-hour session windows with a monthly session limit of 50 sessions.
  - Added a new CLI command to generate session-based usage reports, including detailed metrics, color-coded alerts, and summary statistics.
  - Enhanced reporting options with date filtering, JSON output, cost calculation modes, and customizable sort order.

- **Documentation**
  - Updated the README to include instructions and examples for the new session windows report, CLI usage, and key features.

- **Tests**
  - Added comprehensive tests to validate session window grouping, aggregation, and edge cases.

- **Chores**
  - Reformatted package configuration for consistency (no functional changes).

- **Style**
  - Changed default sort order for reports to show newest data first.

### CODE REVIEW

File: src/commands/sessions.ts [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+				// Month header with utilization stats
+				const utilizationColor = monthStats.utilizationPercent > 80
+					? 'red'
+					: monthStats.utilizationPercent > 60
+						? 'yellow'
+						: 'green';
+
+				logger.info(`\n${pc.bold(monthStats.month)} - ${pc[utilizationColor](`${monthStats.totalSessions} sessions used`)} (${pc[utilizationColor](`${monthStats.utilizationPercent.toFixed(1)}%`)} of 50 limit)`);
+
+				if (monthStats.remainingSessions <= 5 && monthStats.remainingSessions > 0) {
```
Comment (ryoppippi) [inline src/commands/sessions.ts]: what does `5` mean?

**Comment by ryoppippi** [comment]
@EphemeralSapient Hi
Thank you for this PR. Awesome.

I found that there is a magic number `remainingSessions`. 50 session limit is only for $100 plan and $200 plan has different limitation, right? If I'm wrong explain it.

File: src/types.internal.ts [hunk 2]
```diff
@@ -11,3 +11,45 @@ export type CostMode = TupleToUnion<typeof CostModes>;
...
+	utilizationPercent: v.number(), // (totalSessions / 50) * 100
+	totalCost: v.number(),
+	totalTokens: v.number(),
+	averageCostPerSession: v.number(),
+	averageTokensPerSession: v.number(),
+	windows: v.array(SessionWindowSchema),
+	currentSession: v.optional(CurrentSessionInfoSchema),
+});
+
+export type SessionWindowStats = v.InferOutput<typeof SessionWindowStatsSchema>;
```
Comment (ryoppippi) [inline src/types.internal.ts]: you can define those types in `data-loader.ts` because we don't use it in other files.
also you can stop exporting it

**Comment by EphemeralSapient** [comment]
ah, forgot about that completely, I thought both 100$ and 200$ plan had same session; I was wrong

followed up with above review; again used CC , sorry if I missed anything :P

**Comment by ryoppippi** [comment]
@EphemeralSapient I came up an idea.
Instead of creating new sub-command, how about add `--current` option to `session` sub-command?

**Comment by EphemeralSapient** [comment]
after using it a while; I found inconsistent values on window start timing, I didn't notice it earlier

Honestly I'm not sure what to do here 😖

**Comment by ryoppippi** [comment]
> after using it a while; I found inconsistent values on window start timing, I didn't notice it earlier

OMG 

Thank you for investigating.

Well if you cannot make sure that it is correct, we gonna suspend this PR. but thank you for you work

**Comment by EphemeralSapient** [comment]
sorry for the mess, I'll be closing this PR

Have a nice day
